shared/util_linux.go

```diff
@@ -93,8 +95,21 @@ func IsMountPoint(name string) bool {
 	if err != nil {
 		return false
 	}
+
 	// If the directory has the same device as parent, then it's not a mountpoint.
-	return stat.Sys().(*syscall.Stat_t).Dev != rootStat.Sys().(*syscall.Stat_t).Dev
+	if stat.Sys().(*syscall.Stat_t).Dev == rootStat.Sys().(*syscall.Stat_t).Dev {
+		return false
+	}
+
+	// Btrfs annoyingly uses a different Dev id for different subvolumes on the same mount.
+	// So for btrfs, we require a matching mount entry in mountinfo.
+	fs := unix.Statfs_t{}
+	err = unix.Statfs(name, &fs)
+	if err == nil && fs.Type == filesystemSuperMagicBtrfs {
```

Comment:
@stgraber how did the old storage layer work with this? Or was it a bug?

Reply:
The old storage implementation didn't always have a mountpoint. Now that we do the bind-mount trick to always have a mountpoint, we started hitting this bug.

It's quite possible that there were a few other cases where we'd hit this issue and somehow didn't notice though.